lib/logstash/filters/grok/timeout_enforcer.rb

```diff
@@ -0,0 +1,77 @@
+class LogStash::Filters::Grok::TimeoutEnforcer
```

Comment:
I am fine with leaving this in the plugin for now, but I feel like this utility could be better utilized if placed in logstash-core. what do you think?